client/client_test.go

```diff
@@ -0,0 +1,35 @@
+package client
+
+import (
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestTimeout(t *testing.T) {
+	clientConfig := Configuration{
+		Endpoint: "http://httpbin.org/delay/2?",
```

Comment:
I think depending on the network for tests should be fine. But I will look at the alternative. Having it totally internal is definitely better.